docs/matroschka.md

```diff
@@ -0,0 +1,45 @@
+# Matroschka-Prober
+Matroschka-Prober is a GRE based blackbox network monitoring tool
+
+## How it works
+Matroschka functions by stacking GRE headers on top of one another. Each GRE header represents a node or hop. Upon arrival at the hop, the network device decapsulates the package (using a previously configured filter) and forwards it to the next destination. The final packet uses the UDP protocol to return it to the original sender.
+Matroschka makes it possible to test any arbitrary path in the network, even those that are not the shortest path.
+![network diagram](network.png)
+
+## Key features
+- Packets get decapsulated by the ASIC avoiding using valuable CPU time for monitoring
+- Source IP Addresses can be spoofed within IP Subnets, useful to randomize the values used by ECMP capable devices to hash and route packets
+- Configurable TOS/DSCP values
+- Configurable PPS rates
+- Configurable packet payload sizes
+- Configurable measurement durations
+- Provides metrics on /metrics for Prometheus
+
+## Configuration examples to decapsulate packets
+
+### Junos
+`
+term DECAP {
+    from {
+        destination-address {
+            10.1.0.255/32;
+        }
+        protocol gre;
+    }
+    then {
+        decapsulate gre;
+        }
+    }
+    term ACCEPT {
+        then accept;
+    }
+}
+`
```

Comment:
this should probably be a code block with three backticks to improve readability